plugins/pipelines/index.js

```diff
@@ -22,6 +28,20 @@ const startAllRoute = require('./startAll');
  * @param  {Function} next              Function to call when done
  */
 exports.register = (server, options, next) => {
+    /**
+     * Returns true if the scope includes pipeline and it's pipelineId does not matches
+     * the id of the pipeline, otherwise returns false
+     * @method isInvalidToken
+     * @param  {String} id                     ID of pipeline
+     * @param  {Object} credentials            Credential object from Hapi
+     * @param  {String} credentials.pipelineId ID of pipeline which the token is allowed to access
+     * @param  {String} credentials.scope      Scope whose token is allowed
+     */
+    server.expose('isInvalidToken', (id, credentials) =>
```

Comment:
I think this would be more straightforward/clear if we changed the name to `isValidPipelineToken` and then have it return `credentials.scope.includes('pipeline') && parseInt(id, 10) === parseInt(credentials.pipelineId, 10)`

Reply:
Thanks, it's more clear. I'll fix.

Reply:
If non-pipeline scope is passed,  `isValidPipelineToken` returns false and [this](https://github.com/sonic-screwdriver-cd/screwdriver/blob/f54382028b0a1eedd066bc79947733c8177fff59/plugins/builds/create.js#L47-L48) thows unauthorize error. We can check the scope is not `pipeline` in if statement before calling `isValidPipelineToken`, but it's a 
 little redundant.

So, I change the name to `isValidToken` which returns true when the scope is not pipeline or the scope is pipeline and the token's `pipelineId` matches the id of pipeline.